src/flow/persistence/mod.rs

```diff
@@ -191,21 +215,30 @@ impl GroupCommitQueueSet {
         match self.durability_mode {
             DurabilityMode::DeleteOnExit | DurabilityMode::Permanent => {
                 if !self.files.contains_key(node) {
-                    let file = self.create_file(node);
+                    let file = self.get_or_create_file(node);
                     self.files.insert(node.clone(), file);
                 }
 
                 let mut file = &mut self.files[node].1;
                 {
-                    let data_to_flush: Vec<_> = self.pending_packets[&node]
+                    let entries_to_flush: Vec<_> = self.pending_packets[&node]
                         .iter()
                         .map(|p| match **p {
-                            Packet::Transaction { ref data, .. } |
-                            Packet::Message { ref data, .. } => data,
+                            Packet::Transaction { ref data, .. } => LogEntry {
+                                records: Cow::Borrowed(data),
+                                packet_type: PacketType::Transaction,
```

Comment:
I don't think it is necessary to store the packet type on a per message basis. All writes to a base node should be of the same type (and you can check which it is by seeing if the base node is transactional)

Reply:
Ah, didn't think about that. Good point!